engine/verification/verifier/verifiers.go

```diff
@@ -0,0 +1,207 @@
+package verifier
+
+import (
+	"fmt"
+
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+
+	"github.com/onflow/flow-go/cmd/util/cmd/common"
+	"github.com/onflow/flow-go/engine/execution/computation"
+	"github.com/onflow/flow-go/fvm"
+	"github.com/onflow/flow-go/fvm/initialize"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/verification/convert"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/chunks"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/state/protocol"
+	"github.com/onflow/flow-go/storage"
+	storagepebble "github.com/onflow/flow-go/storage/pebble"
+)
+
+// VerifyLastKHeight verifies the last k sealed blocks by verifying all chunks in the results.
+// It assumes the latest sealed block has been executed, and the chunk data packs have not been
+// pruned.
+func VerifyLastKHeight(k uint64, chainID flow.ChainID, protocolDataDir string, chunkDataPackDir string) error {
+	closer, storages, chunkDataPacks, state, verifier, err := initStorages(chainID, protocolDataDir, chunkDataPackDir)
+	if err != nil {
+		return fmt.Errorf("could not init storages: %w", err)
+	}
+	defer func() {
+		err := closer()
+		if err != nil {
+			log.Error().Err(err).Msg("failed to close storages")
+		}
+	}()
```

Comment:
Do we want to return `closer` error as well?

```suggestion
func VerifyLastKHeight(k uint64, chainID flow.ChainID, protocolDataDir string, chunkDataPackDir string) (err error) {
	closer, storages, chunkDataPacks, state, verifier, err := initStorages(chainID, protocolDataDir, chunkDataPackDir)
	if err != nil {
		return fmt.Errorf("could not init storages: %w", err)
	}
	defer func() {
		closerErr := closer()
		if closerErr != nil {
			err = errors.Join(err, closerErr)
		}
	}()
```